src/eva/core/callbacks/writers/embeddings.py

```diff
@@ -90,8 +90,14 @@ def write_on_batch_end(
             embeddings_buffer, target_buffer = io.BytesIO(), io.BytesIO()
```

Comment:
in line 89 we should replace `local_idx` with `global_idx` to prevent duplicate filenames (which could occur if batch_size < n_samples)